pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -91,33 +91,29 @@ class MetadataError(DNFManagerError):
     """Metadata couldn't be loaded."""
 
 
-class MissingSpecsError(DNFManagerError):
-    """Some packages, groups or modules are missing."""
-
-
-class BrokenSpecsError(DNFManagerError):
-    """Some packages, groups or modules are broken."""
-
-
-class InvalidSelectionError(DNFManagerError):
-    """The software selection couldn't be resolved."""
-
-
 class DNFManager:
     """The abstraction of the DNF base."""
 
     def __init__(self):
         self.__base = None
+        self.__goal = None
+        self.__goal_skip_unavailable = None
+
         # Protect access to _base.repos to ensure that the dictionary is not
         # modified while another thread is attempting to iterate over it. The
         # lock only needs to be held during operations that change the number
         # of repos or that iterate over the repos.
         self._lock = threading.RLock()
+
+        self._transaction = None
         self._ignore_missing_packages = False
         self._ignore_broken_packages = False
         self._download_location = None
         self._md_hashes = {}
         self._enabled_system_repositories = []
+        self._repositories_loaded = False
```

Comment:
Why do we need this safeguard now, whereas before it was not needed? Is load_repository aka configure_repository called more than once now?

Reply:
Because previously we checked other things, such as if `_base.sack` exists, but it's not possible in dnf5. There is a `repos_updated_and_loaded` variable, but it's private.